### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by sourav977**
- created `secretstore_chef_types.go` , added related structs
- created `chef.go` and `chef_test.go` , added `NewClient`, `ValidateStore`, `TestValidateStore` method

### CODE REVIEW

File: pkg/provider/chef/chef.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		BaseURL: chefSpec.BaseURL,
+	})
+	if cerr != nil {
+		return nil, fmt.Errorf(errChefClient, err)
+	}
+	providerchef.chefClient = client
+	return providerchef, nil
+}
+
+// TO be implemented
```
Comment (welshDoug) [inline pkg/provider/chef/chef.go:100]: what does this comment relate to?

File: pkg/provider/chef/chef.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+package chef
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+
+	"github.com/go-chef/chef"
```
Comment (welshDoug) [inline pkg/provider/chef/chef.go:21]:
```suggestion
	"github.com/chef/go-chef"
```
I suggest we switch to the fork of go-chef that is maintained by the chef team, it looks to have more recent and regular releases

**Review by welshDoug — CHANGES_REQUESTED** [review]

File: pkg/provider/chef/chef.go (hunk 2 above)
Comment (sourav977) [reply pkg/provider/chef/chef.go:21]: ok @welshDoug we will update it.

File: pkg/provider/chef/chef.go (hunk 1 above)
Comment (sourav977) [reply pkg/provider/chef/chef.go:100]: the above comment line indicates to properly implement all the methods of `type SecretsClient interface` 
Currently we are just returning required fields, more to be done on next PR.

File: go.mod [hunk 3]
```diff
@@ -231,3 +232,5 @@ require (
 	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
+replace github.com/go-chef/chef v0.28.0 => github.com/chef/go-chef v0.4.5
```
Comment (welshDoug) [inline go.mod:236]: why are we using a replace, why don't we just depend on the other version instead?

**Review by welshDoug — APPROVED** [review]

**Comment by sourav977** [comment]
initial changes are approved, planning to keep this `chef-eso-dev` as base branch to develop on.. 
when all components are designed and tested then will create PR to merge with main branch.
Discarding this PR merge with main branch now.

**Pull request closed without merging**